### DISCUSSION TITLE
Can internal cfwheels core function be overwritten in a plugin?

### DISCUSSION THREAD

**Discussion by dhgassoc** [wheels-dev/wheels#1404] [Ideas]
To get CSRF to work with cfwheels on Adobe-2016 in my browser / development environment, I had to modify ` $setAuthenticityToken() `, an internal use only function.   I had to make a simple change; ` (isAjax() || isHtmx() ) `

Changed ` $setAuthenticityToken() ` in /wheels/controller/csrf.cfm, to;
```
 
/**
 * Returns whether the page is from HTMX.
 *
 * [section: Controller]
 * [category: Miscellaneous Functions]
 */
public boolean function isHtmx() {
	return (StructKeyExists(request.$wheelsHeaders, "HX-Request") &&	(request.$wheelsHeaders["HX-Request"]));
}

public function $setAuthenticityToken() {	
	if (!$isVerifiedRequest() && (isAjax() ||      isHtmx()      ) ) {	
		if (StructKeyExists(request.$wheelsHeaders, "X-CSRF-Token")) {	
			params.authenticityToken = request.$wheelsHeaders["X-CSRF-Token"];	
		}	
	}	
}
 
```

Currently I am using a modified version of /wheels/controller/csrf.cfm in the actual /wheels/ folder.  Is there a safer / better place I can put my customized override version of $setAuthenticityToken() ? 
Where can I override this function, outside of /wheels/ folder?

Can you override a cfwheels **_internal_** function in a plugin? The [guide](https://guides.cfwheels.org/2.5.0/plugins/developing-plugins) does not say if you can or not.   I am not confident enough to create a plugin to do this. Peter - would you be willing to add this to your htmx plugin?

**Comment by bpamiri** [comment]
@dhgassoc I can add this functionality to the HTMX plugin. Is there anything else you are doing that would be nice to add at the plugin level?

**Reply by dhgassoc** [reply]
Thanks. thats great.  No more to add at this time.  

I have two other functions, that after more testing, that we  **_may_** want to add.  These are non-core functions.  They address a few problems with adobe-2016 and CSRF and Htmx. Even though they work, I may not be doing it correctly, so I was going to create a separate discussion, to get opinions of others, once I am finish my limited ability to test in my development environment.   And then I need to test in production using Adobe-2016.  I have not even had time to test these other functions without debugging on my (shared) web server.

**Reply by dhgassoc** [reply]
Actually there was one more thing.  Fix the warning for use with cfwheels version 2.5.

Warnings: | The htmx plugin may be incompatible with this version of Wheels, please look for a compatible version of the plugin
Framework: | CFWheels 					2.5.0
Warnings: 	The htmx plugin may be incompatible with this version of Wheels, please look for a compatible version of the plugin
Framework: 	[CFWheels](https://cfwheels.org/) 2.5.0
